http/src/main/resources/jupyter/gce-init.sh

```diff
@@ -0,0 +1,474 @@
+ pos_file /var/tmp/fluentd.welder.pos
+ read_from_head true
+ tag welder
+</source>
+END
+
+    # Add stack driver configuration for jupyter
+    tee /etc/google-fluentd/config.d/jupyter.conf << END
+<source>
+ @type tail
+ format none
+ path /work/jupyter.log
+ pos_file /var/tmp/fluentd.jupyter.pos
+ read_from_head true
+ tag jupyter
+</source>
+END
+
+    # Add stack driver configuration for user startup and shutdown scripts
+    tee /etc/google-fluentd/config.d/daemon.conf << END
+<source>
+ @type tail
+ format none
+ path /var/log/daemon.log
+ pos_file /var/tmp/fluentd.google.user.daemon.pos
+ read_from_head true
+ tag daemon
+</source>
+END
+
+    service google-fluentd reload
+    
+    # Install env var config
+    if [ ! -z "$CUSTOM_ENV_VARS_CONFIG_URI" ] ; then
+      log 'Copy custom env vars config...'
+      gsutil cp ${CUSTOM_ENV_VARS_CONFIG_URI} /etc
+    fi
+
+    # If any image is hosted in a GCR registry (detected by regex) then
+    # authorize docker to interact with gcr.io.
+    if grep -qF "gcr.io" <<< "${JUPYTER_DOCKER_IMAGE}${RSTUDIO_DOCKER_IMAGE}${PROXY_DOCKER_IMAGE}${WELDER_DOCKER_IMAGE}" ; then
+      log 'Authorizing GCR...'
+      gcloud --quiet auth configure-docker
+    fi
+
+    log 'Starting up the Jupydocker...'
+
+    # Run docker-compose for each specified compose file.
+    # Note the `docker-compose pull` is retried to avoid intermittent network errors, but
+    # `docker-compose up` is not retried since if that fails, something is probably broken
+    # and wouldn't be remedied by retrying
+    COMPOSE_FILES=(-f /etc/`basename ${PROXY_DOCKER_COMPOSE}`)
+    cat /etc/`basename ${PROXY_DOCKER_COMPOSE}`
+    if [ ! -z "$JUPYTER_DOCKER_IMAGE" ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${JUPYTER_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${JUPYTER_DOCKER_COMPOSE}`
+    fi
+    if [ ! -z "$RSTUDIO_DOCKER_IMAGE" ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${RSTUDIO_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${RSTUDIO_DOCKER_COMPOSE}`
+    fi
+    if [ ! -z "$WELDER_DOCKER_IMAGE" ] && [ "$WELDER_ENABLED" == "true" ] ; then
+      COMPOSE_FILES+=(-f /etc/`basename ${WELDER_DOCKER_COMPOSE}`)
+      cat /etc/`basename ${WELDER_DOCKER_COMPOSE}`
+    fi
+
+    docker-compose "${COMPOSE_FILES[@]}" config
+    retry 5 docker-compose "${COMPOSE_FILES[@]}" pull
+    docker-compose "${COMPOSE_FILES[@]}" up -d
+
+    # If Welder is installed, start the service.
+    # See https://broadworkbench.atlassian.net/browse/IA-1026
+    if [ ! -z "$WELDER_DOCKER_IMAGE" ] && [ "$WELDER_ENABLED" == "true" ] ; then
+      log 'Starting Welder (file synchronization service)...'
+      retry 3 docker exec -d ${WELDER_SERVER_NAME} /opt/docker/bin/entrypoint.sh
+    fi
+
+    # Jupyter-specific setup, only do if Jupyter is installed
+    if [ ! -z "$JUPYTER_DOCKER_IMAGE" ] ; then
+      log 'Installing Jupydocker kernelspecs...'
+
+      # Change Python and PySpark 2 and 3 kernel specs to allow each to have its own spark
```

Comment:
I'd probably change this comment to `Install kernelspecs inside the Jupyter container`. The TODO is still valid.